pkg/service/init.go

```diff
@@ -147,14 +149,24 @@ func (a *UdmApp) terminateProcedure() {
 	a.CallServerStop()
 
 	// deregister with NRF
-	problemDetails, err := a.Consumer().SendDeregisterNFInstance()
-	if problemDetails != nil {
-		logger.MainLog.Errorf("Deregister NF instance Failed Problem[%+v]", problemDetails)
-	} else if err != nil {
-		logger.MainLog.Errorf("Deregister NF instance Error[%+v]", err)
-	} else {
-		logger.MainLog.Infof("Deregister from NRF successfully")
+	err := a.Consumer().SendDeregisterNFInstance()
+	if err != nil {
+		switch apiErr := err.(type) {
+		case openapi.GenericOpenAPIError:
+			switch errModel := apiErr.Model().(type) {
+			case NFManagement.DeregisterNFInstanceError:
+				pd := &errModel.ProblemDetails
+				logger.InitLog.Errorf("Deregister NF instance Failed Problem[%+v]", pd)
+			case error:
+				logger.InitLog.Errorf("Deregister NF instance Error[%+v]", err)
+			}
+		case error:
+			logger.InitLog.Errorf("Deregister NF instance Error[%+v]", err)
+		}
 	}
+
+	logger.InitLog.Infof("Deregister from NRF successfully")
```

Comment:
"Deregister from NRF successfully" occur only when `err != nil`

Reply:
I think this is right. "Deregister from NRF successfully" will occur in `err == nil`

Reply:
> "Deregister from NRF successfully" occur only when `err != nil`

I think is err == nil